ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java

```diff
@@ -858,11 +858,11 @@ private DataNodeConfig getDataNodeConfig() {
   @Override
   public void onPartitionBecomeBootstrapFromOffline(String partitionName) {
     try {
-      // 1. take actions in storage manager (add new replica if necessary)
-      PartitionStateChangeListener storageManagerListener =
-          partitionStateChangeListeners.get(StateModelListenerType.StorageManagerListener);
-      if (storageManagerListener != null) {
-        storageManagerListener.onPartitionBecomeBootstrapFromOffline(partitionName);
+      // 1. take actions in bootstrap controller
+      PartitionStateChangeListener bootstrapControllerListener =
+            partitionStateChangeListeners.get(StateModelListenerType.BootstrapControllerListener);
+      if (bootstrapControllerListener != null) {
+        bootstrapControllerListener.onPartitionBecomeBootstrapFromOffline(partitionName);
```

Comment:
We should not merge this to master without unit and integration tests yet. This has break the complete flow for Helix integration during new partition addition and restarts.

Reply:
+ This should be config driven and only enabled for one host at a time.

Reply:
+ we need to finalize, if replication manager has a dependency on FCM, then we might need to include that here.

Reply:
Removing the call to BootstrapController from HelixParticipant for now.